system-addon/lib/TopSitesFeed.jsm

```diff
@@ -118,23 +118,49 @@ this.TopSitesFeed = class TopSitesFeed {
     // Augment the pinned links with any other extra data we have for them already in the store
     const links = this.store.getState().TopSites.rows;
     const pinned = NewTabUtils.pinnedLinks.links;
-    return pinned.map(pinnedLink => (pinnedLink ? Object.assign(links.find(link => link && link.url === pinnedLink.url) || {}, pinnedLink) : pinnedLink));
+    return pinned.map(pinnedLink => (pinnedLink ? Object.assign(links.find(link => link && link.url === pinnedLink.url) || {hostname: shortURL(pinnedLink)}, pinnedLink) : pinnedLink));
```

Comment:
I think this should come after not before `Object.assign({}, pinnedLink, {hostname})`. If the pinned link has a bad hostname (you just changed from tiles to AS) the value returned by shortURL will be overwritten.

Reply:
It would be great to refactor this as well, you really have the flex your mind and eyes to read that line.